### PR TITLE
toni-day5

### PR DESCRIPTION
**Pull request by tonilopezmr**
# Toni's solution for day 5 challenge

## Description

> A simple description of your understanding of the problem may be useful.

## Solution

> Here you should paste the most important parts of your solution and explain them.
You can comment the rest of the code with revision comments.

## Tests battery

> This section can be the most interesting. Here, you can show the tests you have written for checking your solution.
You should explain why and which parts you are testing and how.

Happy Advent of Code!

### CODE REVIEW

File: tonilopezmr/day5/MazeOfJumpInstructions.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  private fun jumpInstI(input: List<Int>): Int {
+    var offset = 0
+    var steps = 0
+    val instructions = input.toMutableList()
+
+    while (offset < input.size) {
+      val aux = instructions[offset]
+      instructions.removeAt(offset)
+      instructions.add(offset, aux + 1)
```
Comment (Cotel) [inline tonilopezmr/day5/MazeOfJumpInstructions.kt]: I'm sure Kotlin has a `set` method for mutable lists. You can combine these 2 lines into one 👍

File: tonilopezmr/day5/MazeOfJumpInstructions.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    return steps
+  }
+
+  private fun jumpInst(input: List<Int>): Int = jumpInst(0, 0, input)
+
+  tailrec private fun jumpInst(offset: Int, step: Int, input: List<Int>): Int {
+    return if (offset >= input.size) step
+    else jumpInst(offset + input[offset], step + 1, input.take(offset) + (input[offset] + 1) + input.takeLast(input.size - (offset + 1)))
+  }
```
Comment (Cotel) [inline tonilopezmr/day5/MazeOfJumpInstructions.kt:30]: This recursion is way lighter than mine. It is logic it takes less half the time 💯

File: tonilopezmr/day5/MazeOfJumpInstructionsPartTwo.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  private fun jumpInst(input: List<Int>): Int = jumpInst(0, 0, input)
+
+  tailrec private fun jumpInst(offset: Int, step: Int, input: List<Int>): Int {
+    return if (offset >= input.size) step
+    else jumpInst(offset + input[offset], step + 1, input.take(offset) + newOffset(input, offset) + input.takeLast(input.size - (offset + 1)))
+  }
+
+  private fun newOffset(input: List<Int>, offset: Int) = if (input[offset] >= 3) input[offset] - 1 else input[offset] + 1
```
Comment (Cotel) [inline tonilopezmr/day5/MazeOfJumpInstructionsPartTwo.kt:32]: By passing this function as a parameter you could have reutilized the same class than for part 1 😸

**Review by Cotel — APPROVED** [review]
Seeing this recursion I may have complicated mine unnecessarily 😅

File: tonilopezmr/day5/MazeOfJumpInstructionsPartTwo.kt (hunk 3 above)
Comment (tonilopezmr) [reply tonilopezmr/day5/MazeOfJumpInstructionsPartTwo.kt:32]: yes !

File: tonilopezmr/day5/MazeOfJumpInstructions.kt (hunk 2 above)
Comment (tonilopezmr) [reply tonilopezmr/day5/MazeOfJumpInstructions.kt:30]: Lol

File: tonilopezmr/day5/MazeOfJumpInstructions.kt (hunk 1 above)
Comment (tonilopezmr) [reply tonilopezmr/day5/MazeOfJumpInstructions.kt]: I was looking for it !! thanks!

Comment (tonilopezmr) [reply tonilopezmr/day5/MazeOfJumpInstructions.kt]: Lol you also can do this: `instructions[offset] = aux + 1` I was trying but never success 🤔 💭  thanks

**Pull request merged**